crates/partition-store/src/partition_store_manager.rs

```diff
@@ -42,14 +48,12 @@ pub enum OpenMode {
 
 #[derive(Clone, Debug)]
 pub struct PartitionStoreManager {
-    lookup: Arc<Mutex<PartitionLookup>>,
+    partition_stores: Arc<Mutex<PartitionStoreLookup>>,
+    persisted_lsns: Arc<PersistedLsnLookup>,
```

Comment:
Why not make this part of `PartitionStoreLookup`? The store lookup is behind a Tokio `Mutex` that's used to guard across async operations like opening stores; we don't need to interfere with that for LSN tracking. It's helpful to centralize this as we have a single DB-wide `EventListener` that emits updates for all partitions.